topics/go/algorithms/data/bst/bst.go

```diff
@@ -0,0 +1,113 @@
+package bst
+
+import "fmt"
+
+// TODO This is currently a draft...
+
+// Node represents the data being stored.
+type Node struct {
+	Data  int
+	left  *Node
+	right *Node
+}
+
+// BST represents a binary search tree.
+type BST struct {
+	root *Node
+}
+
+// New creates a new BST.
+func New() *BST {
+	return &BST{
+		root: nil,
```

Comment:
Don't need to specify zero value fields. So this is usable in its zero value state. This means let's remove the factor function. A use can do this.  var bst BST